src/python/base/utils.py

```diff
@@ -782,16 +787,20 @@ def wait_until_timeout(threads, thread_timeout):
 
 def write_data_to_file(content, file_path, append=False):
   """Writes data to file."""
-  content_string = str(content)
   failure_wait_interval = environment.get_value('FAIL_WAIT')
   file_mode = 'ab' if append else 'wb'
   retry_limit = environment.get_value('FAIL_RETRIES')
 
-  for _ in range(retry_limit):
+  # TODO(mbarbella): Require callers to pass strings or bytes then reduce this
+  # back to the original retry limit.
+  for _ in range(retry_limit + 1):
     try:
       with open(file_path, file_mode) as file_handle:
-        file_handle.write(content_string)
-    except:
+        file_handle.write(content)
+    except TypeError:
+      content = str(content).encode('utf-8')
+      continue
+    except EnvironmentError:
```

Comment:
Where is this EnvironmentError defined ?

Reply:
EnvironmentError is the base class for IOError and OSError (both builtin). Added a comment to this block that should make its purpose clearer.